### PR TITLE
Proposes to add an APM Tooling team & eight new APM buildpacks

### PR DESCRIPTION
**Pull request by dmikusa-pivotal**
## Summary

Proposes to add an APM Tooling team & eight new APM buildpacks 🚀 

[Readable Link](https://github.com/paketo-buildpacks/rfcs/blob/apm-tools/text/0000-apm-tools.md).

### CODE REVIEW

**Comment by joshuatcasey** [comment]
Are there additional dependencies with these "new" buildpacks? Will we need to add those dependencies to the existing dependency framework?

File: text/0000-apm-tools.md
Comment (robdimsdale) [inline text/0000-apm-tools.md:34]: Should we re-name these to conform with https://github.com/paketo-buildpacks/rfcs/pull/223 as we bring them in? Or should we bring them in with their current name and re-name them later? I see pros and cons to each approach.

**Comment by dmikusa-pivotal** [comment]
> Are there additional dependencies with these "new" buildpacks? Will we need to add those dependencies to the existing dependency framework?

Do you mean 3rd party binaries? Like what the buildpack installs? Yes, most of them have agents or other various files that get installed. The current configuration for those files is to pull them from the Internet, as we do with the existing APM tool buildpacks.

File: text/0000-apm-tools.md
Comment (robdimsdale) [reply text/0000-apm-tools.md:34]: Either works for me too.

**Comment by dmikusa-pivotal** [comment]
> > Do you mean 3rd party binaries?
> 
> Yeah that's what I meant. It sounds like there would be no additional Paketo-hosted dependencies for these new APM buildpacks.

Correct. Most of them pull from standard public locations (Maven Central, NPM, etc..), or from the vendor's website. There is one, AppDynamics, where the vendor does not provide a public download. That is being taken from an AWS bucket that the CFF hosts for the v2 buildpacks.

**Comment by ForestEckhardt** [comment]
Is there any way to make an overarching metabuildpack will all of these APMs being optional or is there order differences that would not allow for that?

File: text/0000-apm-tools.md
Comment (ekcasey) [inline text/0000-apm-tools.md:33]: For other cross-cutting concerns like tooling/builders/stacks/utilities we haven't always had representation from each language family. Maybe that is also unnecessary here? I am definitely open to including all interested party but if some teams don't care about direct representation on this team, a smaller group of invested folks makes it easier to get quorum.

File: text/0000-apm-tools.md
Comment (ekcasey) [inline text/0000-apm-tools.md:34]:
```suggestion
3. Create eight new repositories, one for each buildpack to be contributed. The APM Tools team will own these repositories. The repository names will follow the pattern `Paketo <technology> Buildpack`, ex: `Paketo Apache Skywalking Buildpack`, as is the existing convention. GitHub repository names will be the name of the technology, ex: `apache-skywalking`. Buildpack ID and image registry names will match the GitHub repository name, ex: `docker.io/paketobuildpacks/apache-skywalking`.
```

**Comment by fg-j** [comment]
From Working Group 7/19/22:

- Next steps: @dmikusa-pivotal to add more detail about the APM buildpacks' detection criteria
- Open question: Do the makers of the APMs (e.g. NewRelic, Dynatrace) want to be able to contribute to these buildpacks?
    - Note that today, governance doesn't have a way for someone to be a single-buildpack maintainer. This needs another RFC! Perhaps a subteam could have permission to bring in a new maintainer to buildpack(s) within their purview.
    - Getting APM makers involved poses a good marketing opportunity – they can call out the buildpacks in their own ecosystems, if they're excited about getting involved.

File: text/0000-apm-tools.md
Comment (dmikusa-pivotal) [reply text/0000-apm-tools.md:33]: Making this optional. Teams can opt-out.

**Comment by dmikusa-pivotal** [comment]
Detection criteria and actions that each buildpack will take, as well as supported languages, are now included in the RFC.

**Comment by joshuatcasey** [comment]
2022-07-26 Working Group notes: waiting for feedback, see previous comments for updates

**Review by ryanmoran — APPROVED** [review]

**Review by ForestEckhardt — APPROVED** [review]

**Comment by dmikusa-pivotal** [comment]
2022-08-02 WG Notes:

Waiting on final approvals. Please add your approval. We would like to merge this soon.

**Review by robdimsdale — APPROVED** [review]

**Review by pivotal-david-osullivan — APPROVED** [review]

File: text/0000-apm-tools.md
Comment (ekcasey) [inline text/0000-apm-tools.md:34]:
```suggestion
3. Create eight new repositories, one for each buildpack to be contributed. The APM Tools team will own these repositories. The repository names will follow the pattern `Paketo Buildpack for <technology>`, ex: `Paketo Buildpacks for Apache Skywalking`, as is the existing [convention](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0050-buildpack-rename.md). GitHub repository names will be the name of the technology, ex: `apache-skywalking`. Buildpack ID and image registry names will match the GitHub repository name, ex: `docker.io/paketobuildpacks/apache-skywalking`.
```

... given we approved https://github.com/paketo-buildpacks/rfcs/pull/223/files

**Review by ekcasey — APPROVED** [review]

**Comment by davidmirror-ops** [comment]
WG 08-09-2022 Notes:
DM: ok to merge

**Pull request merged**
